tests/src/test/java/org/eclipse/hono/tests/registry/CredentialsHttpIT.java

```diff
@@ -541,8 +543,33 @@ public void testGetAddedCredentialsButWithWrongAuthId(final VertxTestContext con
     private static void assertResponseBodyContainsAllCredentials(final VertxTestContext context, final JsonArray responseBody,
             final List<CommonCredential> expected) {
 
+        assertThat(expected.size()).isEqualTo(responseBody.size());
+
+        responseBody.forEach(credential -> {
+            JsonObject.mapFrom(credential).getJsonArray(CredentialsConstants.FIELD_SECRETS)
+                    .forEach(secret -> {
+                        // each secret should contain an ID.
+                        assertThat(JsonObject.mapFrom(secret)
+                                .getString(RegistryManagementConstants.FIELD_SECRETS_ID)).isNotNull();
+                    });
+        });
+
+        // secrets id were added by registry, strip it so we can compare other fields.
+        responseBody.forEach(credential ->{
+            ((JsonObject) credential).getJsonArray(CredentialsConstants.FIELD_SECRETS)
+                    .forEach(secret -> {
+                        ((JsonObject) secret).remove(RegistryManagementConstants.FIELD_SECRETS_ID);
+                    });
+        });
+
+        // The returned secrets won't contains the hashed password details fields, strip them from the expected values.
         final JsonArray expectedArray = new JsonArray();
-        expected.stream().forEach(credential -> expectedArray.add(JsonObject.mapFrom(credential)));
+        expected.stream().forEach(credential -> {
+            final JsonObject jsonCredential = JsonObject.mapFrom(credential);
+            expectedArray.add(stripHashAndSaltFromPasswordSecret(jsonCredential));
```

Comment:
there is not much left to compare, is there?

Reply:
the credential type, auth-id and all the credential-level fields are still there